go-controller/pkg/ovn/pods.go

```diff
@@ -243,58 +271,32 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 
 	var podMac net.HardwareAddr
 	var podIfAddrs []*net.IPNet
-	var cmds []*goovn.OvnCommand
+	var allOps []ovsdb.Operation
 	var addresses []string
-	var cmd *goovn.OvnCommand
 	var releaseIPs bool
 	needsIP := true
 
 	// Check if the pod's logical switch port already exists. If it
 	// does don't re-add the port to OVN as this will change its
 	// UUID and and the port cache, address sets, and port groups
 	// will still have the old UUID.
-	lsp, err := oc.ovnNBClient.LSPGet(portName)
-	if err != nil && err != goovn.ErrorNotFound && err != goovn.ErrorSchema {
+	lsp := &nbdb.LogicalSwitchPort{Name: portName}
+	err = oc.nbClient.Get(lsp)
+	if err != nil && err != libovsdbclient.ErrNotFound {
 		return fmt.Errorf("unable to get the lsp: %s from the nbdb: %s", portName, err)
 	}
+	if len(lsp.UUID) == 0 {
+		lsp.UUID = libovsdbops.BuildNamedUUID()
+	}
 
 	// Bind the port to the node's chassis; prevents ping-ponging between
 	// chassis if ovnkube-node isn't running correctly and hasn't cleared
 	// out iface-id for an old instance of this pod, and the pod got
 	// rescheduled.
-	opts, err := oc.ovnNBClient.LSPGetOptions(portName)
-	if err != nil && err != goovn.ErrorNotFound {
-		klog.Warningf("Failed to get options for port %s: %v", portName, err)
-	}
-	if opts == nil {
-		opts = make(map[string]string)
+	if lsp.Options == nil {
+		lsp.Options = make(map[string]string)
 	}
-	opts["requested-chassis"] = pod.Spec.NodeName
-
-	if lsp == nil {
-		cmd, err = oc.ovnNBClient.LSPAdd(logicalSwitch, portName)
-		if err != nil {
-			return fmt.Errorf("unable to create the LSPAdd command for port: %s from the nbdb: %v", portName, err)
-		}
-		cmds = append(cmds, cmd)
-		// Unique identifier to distinguish interfaces for recreated pods, also set by ovnkube-node
-		// ovn-controller will claim the OVS interface only if external_ids:iface-id
-		// matches with the Port_Binding.logical_port and external_ids:iface-id-ver matches
-		// with the Port_Binding.options:iface-id-ver. This is not mandatory.
-		// If Port_binding.options:iface-id-ver is not set, then OVS
-		// Interface.external_ids:iface-id-ver if set is ignored.
-		// Only set for new LSP for correct ovn-kube upgrade, because for old OVS Interfaces
-		// iface-id-ver is not set => ovn-controller won't bind OVS Interface
-		opts["iface-id-ver"] = string(pod.UID)
-	} else {
-		klog.Infof("LSP already exists for port: %s", portName)
-	}
-
-	cmd, err = oc.ovnNBClient.LSPSetOptions(portName, opts)
-	if err != nil {
-		return fmt.Errorf("unable to create the LSPSetOptions command for port: %s from the nbdb: %v", portName, err)
-	}
-	cmds = append(cmds, cmd)
+	lsp.Options["requested-chassis"] = pod.Spec.NodeName
```

Comment:
@JacobTanenbaum looks like iface-id-ver got lost in a rebase maybe? Still need to add that to the LSP options if it wasn't already there.

Reply:
My mistake, added it